File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    const map = new Map()
    const target = nums.length / 2
    let ret = -1

    nums.forEach((num) => {
        const upd = getOrDefault(map, num, 0)+1
        if (upd > target) {
            ret = num;
        }
        map.set(num,upd)
    })
    return ret
};

var getOrDefault = function(map, value, def) {
    const temp = map.get(value)

    return temp != undefined ? temp : def;
}

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        final int target = nums.length / 2;
        Map<Integer, Integer> freqs = new HashMap<>(); // k: num, v: freq
        for (int x : nums) {
            int freq = freqs.getOrDefault(x, 0);
            if (freq+1 > target) {return x;}
            else {freqs.put(x, freq+1);}
        }
        return -1;
    }
}